eng/validation/build-arcadewithrepo.ps1

```diff
@@ -45,64 +44,32 @@ $testRootBase = if ($env:AGENT_WORKFOLDER) { $env:AGENT_WORKFOLDER } else { $([S
 $testRoot = Join-Path -Path $testRootBase -ChildPath $([System.IO.Path]::GetRandomFileName())
 New-Item -Path $testRoot -ItemType Directory | Out-Null
 
-$global:minTime = (Get-Date).AddDays(-$global:daysOfOldestBuild)
-$global:buildReasonsList = @("batchedCI", "individualCI", "manual")
-$global:buildSuccessResultList = @("succeeded", "partiallySucceeded")
-
-function Get-LastKnownGoodBuildSha()
+function Get-LatestBuildSha()
 {
-    ## Have there been any builds in the last $daysOfOldestBuild days? 
+    ## Verified that this API gets completed builds, not in progress builds
     $headers = Get-AzDOHeaders
-    $count = 0
+    $uri = "https://dev.azure.com/${global:azdoOrg}/${global:azdoProject}/_apis/build/latest/${global:buildDefinitionId}?branchName=${global:subscribedBranchName}&api-version=5.1-preview.1"
+    $response = (Invoke-WebRequest -Uri $uri -Headers $headers -Method Get) | ConvertFrom-Json
 
-    foreach($reason in $global:buildReasonsList)
+    ## Is it successful or partially successful? Then use that as the foundation for our branch. 
+    ## How do we want to handle cancelled builds? 
```

Comment:
I think we already have the answer to this question in the code below: we handle them as not successful

Reply:
I'll take that note to self out of there 😂